wiki/python/ESTIMATOR_GUIDE.md

```diff
@@ -0,0 +1,866 @@
+# cuML Python Estimators Developer Guide
+
+This guide is meant to help developers follow the correct patterns when creating/modifying any cuML Estimator object and ensure a uniform cuML API.
+
+**Note:** This guide is long, because it includes internal details on how cuML manages input and output types for advanced use cases. But for the vast majority of estimators, the requirements are very simple and can follow the example patterns shown below in the [Quick Start Guide](#quick-start-guide).
+
+Start by reading the following:
+
+1. [SkLearn Estimator Docs](https://scikit-learn.org/stable/developers/develop.html)
+   1. cuML Estimator design follows Sklearn very closely. We will only cover portions where our design differs from this document
+   2. Pay close attention to these sections as these topics have caused pain points in the past:
+      1. [Instantiation](https://scikit-learn.org/stable/developers/develop.html#estimated-attributes)
+      2. [Estimated Attributes](https://scikit-learn.org/stable/developers/develop.html#estimated-attributes)
+      3. [`get_params` and `set_params`](https://scikit-learn.org/stable/developers/develop.html#estimated-attributes)
+      4. [cloning](https://scikit-learn.org/stable/developers/develop.html#estimated-attributes)
+2. [SkLearn Docstring Guide](https://scikit-learn.org/stable/developers/contributing.html#guidelines-for-writing-documentation)
+   1. We follow the same guidlines for speficying array-like objects, array shapes, dtypes, and default values
+
+## Quick Start Guide
+
+At a high level, all cuML Estimators must:
```

Comment:
```suggestion
At a high level, all standard cuML Estimators must:
```
We should add something to avoid confusion when things from `preprocessing` or other parts that don't need to inherit from `Base`. This is consistent with how afterwards it states `For the majority of estimators, the above steps...`

Reply:
Not sure I follow your suggestion. If something doesn't inherit from `Base`, it wouldnt be considered an estimator. Are you referring to some of the `preprocessing` classes that dont exactly fit the estimator mold?

Reply:
Discussed with Dante. Resolving